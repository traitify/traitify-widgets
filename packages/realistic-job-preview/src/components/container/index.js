import PropTypes from "prop-types";
import Context from "lib/context";
import ErrorBoundary from "./error-boundary";
import Theme from "./theme";
import useReducer from "./use-reducer";

function Container({children, className = null, ...props}) {
  const state = useReducer({...props});

  return (
    <Context.Provider value={state}>
      <Theme className={className}>
        <ErrorBoundary>{children}</ErrorBoundary>
      </Theme>
    </Context.Provider>
  );
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

export default Container;
