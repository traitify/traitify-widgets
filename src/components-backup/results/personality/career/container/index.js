import PropTypes from "prop-types";
import TraitifyPropTypes from "lib/helpers/prop-types";
import useDidMount from "lib/hooks/use-did-mount";
import useDidUpdate from "lib/hooks/use-did-update";
import withTraitify from "lib/with-traitify";
import CareerFilter from "../filter";
import CareerList from "../list";
import CareerModal from "../modal";
import style from "./style.scss";

function CareerContainer(props) {
  const {ui, isReady, setElement} = props;
  const state = {};

  useDidMount(() => { ui.trigger("Careers.initialized", {props, state}); });
  useDidUpdate(() => { ui.trigger("Careers.updated", {props, state}); });

  return (
    isReady("results") && (
      <div className={style.container} ref={setElement}>
        <CareerFilter {...props} />
        <CareerList {...props} />
        <CareerModal {...props} />
      </div>
    )
  );
}
CareerContainer.propTypes = {
  isReady: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired,
  setElement: PropTypes.func.isRequired
};

export {CareerContainer as Component};
export default withTraitify(CareerContainer);
