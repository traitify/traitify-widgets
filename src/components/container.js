/* eslint-disable react/forbid-prop-types */
import PropTypes from "prop-types";
import {useEffect} from "react";
import {RecoilRoot, useRecoilState, useSetRecoilState} from "recoil";
import slice from "lib/common/object/slice";
import Http from "lib/http";
import I18n from "lib/i18n";
import Listener from "lib/listener";
import {
  assessmentIDState,
  benchmarkIDState,
  httpState,
  i18nState,
  listenerState,
  loadingState,
  localeState,
  optionsState,
  profileIDState
} from "lib/recoil";

function State({children, http: _http, i18n: _i18n, listener: _listener, options}) {
  const setAssessmentID = useSetRecoilState(assessmentIDState);
  const setBenchmarkID = useSetRecoilState(benchmarkIDState);
  const setHttp = useSetRecoilState(httpState);
  const setI18n = useSetRecoilState(i18nState);
  const setListener = useSetRecoilState(listenerState);
  const setLocale = useSetRecoilState(localeState);
  const setOptions = useSetRecoilState(optionsState);
  const setProfileID = useSetRecoilState(profileIDState);
  const [loading, setLoading] = useRecoilState(loadingState);
  const {
    assessmentID,
    benchmarkID,
    locale,
    profileID,
    ...extraOptions
  } = options;

  useEffect(() => {
    setOptions(extraOptions);
  }, [options]);

  useEffect(() => {
    if(!locale) { return; }

    setLocale(locale);
  }, [locale]);

  useEffect(() => {
    const http = _http || new Http(slice(options, ["authKey", "host", "version"]));

    setHttp(http);
  }, [_http]);

  useEffect(() => {
    const i18n = _i18n || new I18n();

    setI18n(i18n);
  }, [_i18n]);

  useEffect(() => {
    const listener = _listener || new Listener();

    setListener(listener);
  }, [_listener]);

  useEffect(() => {
    setAssessmentID(assessmentID);
  }, [assessmentID]);

  useEffect(() => {
    setBenchmarkID(benchmarkID);
  }, [benchmarkID]);

  useEffect(() => {
    setProfileID(profileID);
  }, [profileID]);

  useEffect(() => {
    setLoading(false);
  }, []);

  if(loading) { return null; }

  return children;
}

State.defaultProps = {
  http: null,
  i18n: null,
  listener: null,
  options: {}
};
State.propTypes = {
  children: PropTypes.node.isRequired,
  http: PropTypes.object,
  i18n: PropTypes.object,
  listener: PropTypes.object,
  options: PropTypes.shape({
    assessmentID: PropTypes.string,
    benchmarkID: PropTypes.string,
    locale: PropTypes.string,
    profileID: PropTypes.string
  })
};

export default function Container(props) {
  return (
    <RecoilRoot>
      <State {...props} />
    </RecoilRoot>
  );
}
