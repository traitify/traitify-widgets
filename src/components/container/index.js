import {RecoilRoot} from "recoil";
import ErrorBoundary from "./error-boundary";
import State from "./state";

export default function Container(props) {
  return (
    <RecoilRoot>
      <ErrorBoundary>
        <State {...props} />
      </ErrorBoundary>
    </RecoilRoot>
  );
}
