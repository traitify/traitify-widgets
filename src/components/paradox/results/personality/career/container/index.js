import PropTypes from "prop-types";
import {useEffect} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import CareerFilter from "../filter";
import CareerModal from "../modal";
import CareerResults from "../list";
import style from "./style.scss";

function Careers(props) {
  const {ui, isReady} = props;

  useEffect(() => {
    ui.trigger("Careers.initialized", {props});
  }, []);
  useEffect(() => {
    ui.trigger("Careers.updated", {props});
  });

  return (
    isReady("results") && (
      <div className={style.container}>
        <CareerFilter {...props} />
        <CareerResults {...props} />
        <CareerModal {...props} />
      </div>
    )
  );
}
Careers.propTypes = {
  isReady: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};
export {Careers as Component};
export default withTraitify(Careers);
