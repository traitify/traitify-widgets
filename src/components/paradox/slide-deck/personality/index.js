import {
  faChevronLeft,
  faCompressArrowsAlt,
  faExpandArrowsAlt
} from "@fortawesome/free-solid-svg-icons";
import Markdown from "markdown-to-jsx";
import PropTypes from "prop-types";
import {useEffect, useLayoutEffect, useRef, useState} from "react";
import Loading from "components/loading";
import Icon from "lib/helpers/icon";
import {dig, slice, toQueryString} from "lib/helpers/object";
import TraitifyPropTypes from "lib/helpers/prop-types";
import {camelCase} from "lib/helpers/string";
import useDidMount from "lib/hooks/use-did-mount";
import useDidUpdate from "lib/hooks/use-did-update";
import useFullscreen from "lib/hooks/use-fullscreen";
import useSlideLoader from "lib/hooks/use-slide-loader";
import withTraitify from "lib/with-traitify";
import Image from "./image";
import style from "./style.scss";

const maxRetries = 2;

function Personality({element, setElement, ...props}) {
  const {
    assessment,
    cache,
    getAssessment,
    getCacheKey,
    getOption,
    isReady,
    traitify,
    translate,
    ui
  } = props;
  const caption = useRef(null);
  const image = useRef(null);
  const {
    error,
    dispatch,
    ready,
    slideIndex,
    slides
  } = useSlideLoader({element: image.current, translate});
  const [fullscreen, toggleFullscreen] = useFullscreen(element);
  const [instructions, setInstructions] = useState(null);
  const [showInstructions, setShowInstructions] = useState(false);
  const [submitAttempts, setSubmitAttempts] = useState(0);
  const [submitError, setSubmitError] = useState(null);
  const submitting = useRef(false);

  const completedSlides = slides.filter(({response}) => response != null)
    .map((slide) => slice(slide, ["id", "response", "time_taken"]));
  const finished = slides.length > 0 && slides.length === completedSlides.length;
  const likertScale = dig(assessment, "scoring_scale") === "LIKERT_CUMULATIVE_POMP";
  const resultsReady = isReady("results");
  const state = {
    finished,
    fullscreen,
    showInstructions,
    slides
  };

  useDidMount(() => { ui.trigger("SlideDeck.initialized", {props, state}); });
  useDidUpdate(() => { ui.trigger("SlideDeck.updated", {props, state}); });
  useDidUpdate(() => { ui.trigger("SlideDeck.fullscreen", {props, state}, fullscreen); }, [fullscreen]);
  useEffect(() => {
    if(!dig(assessment, "slides")) { return; }

    const cachedData = cache.get(getCacheKey("slide-deck")) || {};
    const getImageURL = ({size, slide}) => {
      const imageHost = getOption("imageHost");
      const [width, height] = size;
      if(width <= 0 || height <= 0) { return slide.image_desktop; }

      const baseURL = `${imageHost}/slides/${slide.image_desktop_retina.split("/").pop()}`;
      const params = {
        "auto": "format",
        "fp-x": slide.focus_x / 100,
        "fp-y": slide.focus_y / 100,
        "h": (likertScale && window.innerWidth <= 768) ? height - 74 : height,
        "w": width
      };

      return `${baseURL}?${toQueryString(params)}`;
    };

    dispatch({
      cachedSlides: cachedData.slides || [],
      getImageURL,
      slides: assessment.slides,
      type: "reset"
    });
  }, [
    dig(assessment, "id"),
    dig(assessment, "locale_key"),
    dig(assessment, "slides")
  ]);

  useEffect(() => {
    const show = getOption("allowInstructions");
    const text = dig(assessment, "instructions", "instructional_text");

    setInstructions(text);
    setShowInstructions(show && !!text);
  }, [dig(assessment, "instructions")]);

  useEffect(() => {
    if(completedSlides.length === 0) { return; }

    cache.set(getCacheKey("slide-deck"), {slides: completedSlides});
  }, [completedSlides]);

  useEffect(() => {
    if(!finished) { return; }
    if(resultsReady) { return; }
    if(submitAttempts > maxRetries) { return; }
    if(submitting.current) { return; }

    submitting.current = true;

    traitify.put(
      `/assessments/${assessment.id}/slides`,
      completedSlides.map(({id, response, time_taken: timeTaken}) => ({
        id,
        [likertScale ? "likert_response" : "response"]: response,
        time_taken: timeTaken && timeTaken >= 0 ? timeTaken : 2
      }))
    ).then((response) => {
      submitting.current = false;

      ui.trigger("SlideDeck.finished", {state, props}, response);
      getAssessment({force: true});
    }).catch((response) => {
      submitting.current = false;

      let newError;

      try {
        newError = JSON.parse(response).errors[0];
      } catch(e) {
        newError = response;
      }

      setTimeout(() => setSubmitAttempts((x) => x + 1), 2000);
      if(submitAttempts >= maxRetries) { setSubmitError(newError || "Error Submitting"); }
    });
  }, [finished, submitAttempts]);

  useLayoutEffect(() => {
    if(!caption.current) { return; }

    caption.current.focus();
  }, [showInstructions, slideIndex]);

  if(resultsReady) { return null; }

  const back = () => dispatch({type: "back"});
  const content = {};
  const currentSlide = slides[slideIndex];
  const updateSlide = (index, value) => {
    dispatch({index, response: value, type: "answer"});

    const key = likertScale ? `likert.${camelCase(value)}` : camelCase(value ? "me" : "not-me");
    ui.trigger(`SlideDeck.${key}`, {props, state});
    ui.trigger("SlideDeck.updateSlide", {props, state});
  };

  if(submitError || error) {
    const retry = () => {
      if(!submitError) { return dispatch({type: "retry"}); }

      setSubmitAttempts(0);
      setSubmitError(null);
    };

    content.caption = submitError || error;
    content.image = (
      <div className={style.error}>
        <button className={style.link} onClick={retry} type="button">
          {translate("try_again")}
        </button>
      </div>
    );
  } else if(showInstructions) {
    content.caption = translate("instructions");
    content.image = (
      <>
        <div className={`${style.slide} ${style.middle}`}>
          <div className={style.instructionsSlide}>
            <div className={style.instructionsText}>
              <Markdown>{instructions}</Markdown>
            </div>
            <button className={style.instructionsButton} onClick={() => setShowInstructions(false)} type="button">
              {translate("get_started")}
            </button>
          </div>
        </div>
        {currentSlide && <Image key={currentSlide.id} orientation="right" slide={currentSlide} />}
      </>
    );
  } else if(!ready || finished) {
    content.caption = translate("loading");
    content.image = <div className={style.loading}><Loading /></div>;
  } else {
    const lastSlide = slides[slideIndex - 1];
    const nextSlide = slides[slideIndex + 1];
    const progress = slides.length > 0 ? (slideIndex / slides.length) * 100 : 0;

    content.caption = currentSlide.caption;
    content.enabled = true;
    content.image = (
      <>
        <div className={style.progress} style={{width: `${progress}%`}} />
        {lastSlide && <Image key={lastSlide.id} orientation="left" slide={lastSlide} />}
        <Image key={currentSlide.id} orientation="middle" slide={currentSlide} />
        {nextSlide && <Image key={nextSlide.id} orientation="right" slide={nextSlide} />}
        {getOption("allowBack") && slideIndex > 0 && (
          <button className={style.back} onClick={back} type="button">
            <Icon alt={translate("back")} icon={faChevronLeft} />
          </button>
        )}
        {getOption("allowFullscreen") && (
          <button className={style.fullscreen} onClick={toggleFullscreen} type="button">
            <Icon icon={fullscreen ? faCompressArrowsAlt : faExpandArrowsAlt} />
          </button>
        )}
      </>
    );
  }

  const buttonClass = [
    "traitify--response-button",
    style.response,
    !content.enabled && style.btnDisabled
  ].filter(Boolean).join(" ");
  const buttons = likertScale ? [
    {key: "really_not_me", response: "REALLY_NOT_ME"},
    {key: "not_me", response: "NOT_ME"},
    {key: "me", response: "ME"},
    {key: "really_me", response: "REALLY_ME"}
  ] : [
    {key: "me", response: true},
    {key: "not_me", response: false}
  ];

  return (
    <div className={[style.container, likertScale && style.likertScale].filter(Boolean).join(" ")} ref={setElement}>
      <div className={style.caption} ref={caption} tabIndex="-1">{content.caption}</div>
      <div className={style.image} ref={image}>{content.image}</div>
      <div className={style.buttons}>
        {buttons.map(({key, response}) => (
          <button
            key={key}
            className={[buttonClass, style[camelCase(key)]].join(" ")}
            onClick={content.enabled && (() => updateSlide(slideIndex, response))}
            type="button"
          >
            {translate(key)}
          </button>
        ))}
      </div>
    </div>
  );
}

Personality.defaultProps = {assessment: null, element: null};
Personality.propTypes = {
  assessment: PropTypes.shape({
    id: PropTypes.string.isRequired,
    instructions: PropTypes.shape({instructional_text: PropTypes.string.isRequired}),
    locale_key: PropTypes.string.isRequired,
    scoring_scale: PropTypes.string,
    slides: PropTypes.arrayOf(PropTypes.object).isRequired
  }),
  cache: PropTypes.shape({
    get: PropTypes.func.isRequired,
    set: PropTypes.func.isRequired
  }).isRequired,
  element: PropTypes.instanceOf(Element),
  getAssessment: PropTypes.func.isRequired,
  getCacheKey: PropTypes.func.isRequired,
  getOption: PropTypes.func.isRequired,
  isReady: PropTypes.func.isRequired,
  setElement: PropTypes.func.isRequired,
  traitify: TraitifyPropTypes.traitify.isRequired,
  translate: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};

export {Personality as Component};
export default withTraitify(Personality);
