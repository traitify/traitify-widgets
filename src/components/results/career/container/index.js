import useComponentEvents from "lib/hooks/use-component-events";
import useResults from "lib/hooks/use-results";
import style from "./style.scss";
import CareerFilter from "../filter";
import CareerList from "../list";
import CareerModal from "../modal";

export default function CareerContainer() {
  const results = useResults({surveyType: "personality"});

  useComponentEvents("Careers");

  if(!results) { return null; }

  return (
    <div className={style.container}>
      <CareerFilter />
      <CareerList />
      <CareerModal />
    </div>
  );
}
