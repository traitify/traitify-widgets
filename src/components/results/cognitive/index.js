import DangerousHTML from "components/common/dangerous-html";
import useTranslate from "lib/hooks/use-translate";
import style from "./style.scss";

export default function Cognitive() {
  const translate = useTranslate();

  return (
    <section className={style.container}>
      <img alt="Brain" src="https://cdn.traitify.com/images/cognitive/brain.png" />
      <div className={style.heading}>{translate("cognitive_results_heading")}</div>
      <DangerousHTML html={translate("cognitive_results_html")} />
    </section>
  );
}
