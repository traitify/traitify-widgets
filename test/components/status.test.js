import {act} from "react-test-renderer";
import Component from "components/status";
import ComponentHandler from "support/component-handler";
import {mockAssessments, mockCognitiveAssessment} from "support/container/http";
import {mockOption} from "support/container/options";
import useContainer from "support/hooks/use-container";
import cognitive from "support/json/assessment/cognitive.json";
import personality from "support/json/assessment/dimension-based.json";

const external = {
  assessmentId: "external-id",
  link: "https://external.traitify.com/external-assessment-id",
  status: "INCOMPLETE",
  surveyId: "external-assessment-id",
  surveyName: "Emmersion Assessment",
  vendor: "Emmersion"
};

const responseToArray = (response) => {
  const assessments = [];
  const {
    cognitive: cognitiveAssessment,
    external: externalAssessment,
    personality: personalityAssessment
  } = response.prerequisites || {};

  if(personalityAssessment && personalityAssessment.assessmentId) {
    assessments.push({
      completed: personalityAssessment.status === "COMPLETE",
      id: personalityAssessment.assessmentId,
      name: "Personality Assessment",
      type: "personality"
    });
  }

  if(cognitiveAssessment && cognitiveAssessment.testId) {
    assessments.push({
      completed: cognitiveAssessment.status === "COMPLETE",
      id: cognitiveAssessment.testId,
      name: "Cognitive Assessment",
      type: "cognitive"
    });
  }

  if(externalAssessment) {
    externalAssessment.forEach((assessment) => {
      assessments.push({
        completed: assessment.status === "COMPLETE",
        id: assessment.assessmentId,
        link: assessment.link,
        name: assessment.surveyName,
        type: "external"
      });
    });
  }

  return assessments;
};

describe("Status", () => {
  let assessments;
  let assessmentsResponse;
  let component;

  useContainer();

  beforeEach(() => {
    assessmentsResponse = {
      benchmarkID: "benchmark-id",
      prerequisites: {
        cognitive: {
          status: "COMPLETE",
          surveyId: cognitive.surveyId,
          testId: cognitive.id
        },
        external: [{...external}],
        personality: {
          assessmentId: personality.id,
          status: "COMPLETE",
          surveyId: personality.deck_id
        }
      },
      profileID: "profile-id"
    };
    assessments = responseToArray(assessmentsResponse);

    mockAssessments(assessmentsResponse);
  });

  describe("callbacks", () => {
    it("triggers initialization", async() => {
      await ComponentHandler.setup(Component);

      expect(container.listener.trigger).toHaveBeenCalledWith(
        "Status.initialized",
        {assessments: null}
      );
    });

    it("triggers update", async() => {
      component = await ComponentHandler.setup(Component);
      await component.update();

      expect(container.listener.trigger).toHaveBeenCalledWith(
        "Status.updated",
        {assessments}
      );
    });
  });

  it("starts assessment", async() => {
    assessmentsResponse.prerequisites.cognitive.status = "INCOMPLETE";
    assessments = responseToArray(assessmentsResponse);
    mockCognitiveAssessment(cognitive);
    mockAssessments(assessmentsResponse);
    component = await ComponentHandler.setup(Component);
    const button = component.findAllByText("Start Assessment")[0];
    act(() => button.props.onClick());

    expect(container.listener.trigger).toHaveBeenCalledWith(
      "Survey.start",
      {assessment: assessments[1]}
    );
  });

  it("starts external assessment", async() => {
    mockOption("status", {allowRedirect: false});
    component = await ComponentHandler.setup(Component);
    const button = component.findByText("Start Assessment");
    act(() => button.props.onClick());

    expect(container.listener.trigger).toHaveBeenCalledWith(
      "Survey.start",
      {assessment: assessments[2]}
    );
  });

  it("starts external assessment with redirect", async() => {
    component = await ComponentHandler.setup(Component);
    const button = component.findByText("Start Assessment");

    expect(button.props.href).toBe(external.link);
  });

  it("renders component", async() => {
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if assessments not ready", async() => {
    mockAssessments({implementation: () => new Promise(() => {})});
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if no assessments", async() => {
    assessmentsResponse.prerequisites = {};
    assessments = responseToArray(assessmentsResponse);
    mockAssessments(assessmentsResponse);
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });
});
