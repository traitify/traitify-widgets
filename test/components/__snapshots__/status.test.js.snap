// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status order renders component 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <div
    className="container"
  >
    <div
      className="header"
    >
      Your Application Assessments
    </div>
    <div
      className="p"
    >
      As part of your application, we'd like to ask you to complete the following assessments. Please click on the button next to the assessment name. This will take you to where you'll complete the assessment and either be returned to this page if you have multiple assessments to complete or taken to the next stage of the process.
    </div>
    <div
      className="assessment inactive"
    >
      <div>
        <span
          className="icon traitify--icon-fa"
          dangerouslySetInnerHTML={
            {
              "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="check" class="svg-inline--fa fa-check" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7 393.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"></path></svg>",
            }
          }
        />
        <span
          className="text"
        >
          Cognitive Assessment
        </span>
      </div>
      <button
        disabled={true}
        type="button"
      >
        Complete
      </button>
    </div>
    <div
      className="assessment"
    >
      <div>
        <span
          className="text"
        >
          Emmersion Assessment
        </span>
      </div>
      <a
        href="https://external.traitify.com/external-assessment-id"
      >
        Start Assessment
      </a>
    </div>
    <div
      className="assessment inactive"
    >
      <div>
        <span
          className="icon traitify--icon-fa"
          dangerouslySetInnerHTML={
            {
              "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="check" class="svg-inline--fa fa-check" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7 393.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"></path></svg>",
            }
          }
        />
        <span
          className="text"
        >
          Personality Assessment
        </span>
      </div>
      <button
        disabled={true}
        type="button"
      >
        Complete
      </button>
    </div>
  </div>
</div>
`;

exports[`Status order renders error 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <div
    className="container"
  >
    <div
      className="header"
    >
      Your Application Assessments
    </div>
    <div
      className="p"
    >
      There was an error with your assessments
    </div>
    <button
      className="retry"
      onClick={[Function]}
      type="button"
    >
      Let's Try Again
    </button>
  </div>
</div>
`;

exports[`Status order renders loading 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <div
    className="container"
  >
    <img
      alt=""
      src="https://cdn.traitify.com/widgets/status/loading-sonar.gif"
    />
    <div
      className="bold p"
    >
      Loading your assessment
    </div>
  </div>
</div>
`;

exports[`Status order renders skipped 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <section
    className="container"
  >
    <div
      className="header"
    >
      Accommodation Request Submitted
    </div>
    <div
      className="p"
      dangerouslySetInnerHTML={
        {
          "__html": "A member of our team will review your information as soon as possible",
        }
      }
    />
  </section>
</div>
`;

exports[`Status recommendation renders component 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <div
    className="container"
  >
    <div
      className="header"
    >
      Your Application Assessments
    </div>
    <div
      className="p"
    >
      As part of your application, we'd like to ask you to complete the following assessments. Please click on the button next to the assessment name. This will take you to where you'll complete the assessment and either be returned to this page if you have multiple assessments to complete or taken to the next stage of the process.
    </div>
    <div
      className="assessment inactive"
    >
      <div>
        <span
          className="icon traitify--icon-fa"
          dangerouslySetInnerHTML={
            {
              "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="check" class="svg-inline--fa fa-check" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7 393.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"></path></svg>",
            }
          }
        />
        <span
          className="text"
        >
          Personality Assessment
        </span>
      </div>
      <button
        disabled={true}
        type="button"
      >
        Complete
      </button>
    </div>
    <div
      className="assessment inactive"
    >
      <div>
        <span
          className="icon traitify--icon-fa"
          dangerouslySetInnerHTML={
            {
              "__html": "<svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="check" class="svg-inline--fa fa-check" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M438.6 105.4c12.5 12.5 12.5 32.8 0 45.3l-256 256c-12.5 12.5-32.8 12.5-45.3 0l-128-128c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L160 338.7 393.4 105.4c12.5-12.5 32.8-12.5 45.3 0z"></path></svg>",
            }
          }
        />
        <span
          className="text"
        >
          Cognitive Assessment
        </span>
      </div>
      <button
        disabled={true}
        type="button"
      >
        Complete
      </button>
    </div>
    <div
      className="assessment"
    >
      <div>
        <span
          className="text"
        >
          Emmersion Assessment
        </span>
      </div>
      <a
        href="https://external.traitify.com/external-assessment-id"
      >
        Start Assessment
      </a>
    </div>
  </div>
</div>
`;

exports[`Status recommendation renders nothing if assessments not ready 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
/>
`;

exports[`Status recommendation renders nothing if no assessments 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
/>
`;
